src/backend/renderer/mod.rs

```diff
@@ -168,26 +168,62 @@ pub trait Frame {
         &mut self,
         texture: &Self::TextureId,
         matrix: Matrix3<f32>,
+        verticies: [Vector2<f32>; 4],
         alpha: f32,
     ) -> Result<(), Self::Error>;
+
     /// Render a texture to the current target as a flat 2d-plane at a given
-    /// position, applying the given transformation with the given alpha value.
+    /// position, cropping at the given offsets and applying the given transformation with the given alpha value.
     ///
     /// This operation is only valid in between a `begin` and `finish`-call.
     /// If called outside this operation may error-out, do nothing or modify future rendering results in any way.
     fn render_texture_at(
         &mut self,
         texture: &Self::TextureId,
         pos: (i32, i32),
+        scale: u32,
```

Comment:
What happens if the compositor wishes to draw a texture whose buffer scale is 2 on an output whose scaling factor is 3?  Should it use `render_texture_from_to` instead? If yes, that might be worth mentioning in the docs.

Reply:
Oh right, so we do not some kind of fractional scaling for mixing and matching different output and buffer scales, that are not multiples of each other, I was not really aware of that. I wonder, if the better solution would be to allow a floating point scale here.

Reply:
That, or provide the buffer scale as argument as well as the desired output scale, I'd say.